lapce-proxy/src/plugin/mod.rs

```diff
@@ -1375,6 +1375,8 @@ pub fn remove_volt(
                 .volt_removing(volt.clone(), "Plugin Directory not set".to_string());
             anyhow::anyhow!("don't have dir")
         })?;
+        // Waiting for the instance of WASI to be dropped.
+        std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
It would be much better to wait for the wasi instance to be dropped in `stop_volt()` I think.

My idea is in `PluginServerRpcHandler::shutdown()`, you'd wait for a new event like `PluginServerRpc::Exited` which was sent from the wasi msg sending loop break. (wasi.rs line 464)

Reply:
Okay, let me optimize it tomorrow.

Reply:
I've reorganized the code and have a few questions. 1. Can `PluginServerRpcHandler::mainloop` not delete the plugin folder because it can't access class `PluginCatalogRpcHandler`? 2. After calls `PluginServerRpcHandler::shutdown()`, `PluginServerRpcHandler::mainloop` can no longer accept new events, nor can it send notifications and requests to `PluginCatalogRpcHandler`? in addition, I think the most clear and straightforward way is for `ProxyRpcHandler` to send a close request directly, and then delete the folder in the callback. What do you think?

Reply:
You are right. Deleting in a callback looks to be the best solution.